components/steps/icons.tsx

```diff
@@ -0,0 +1,150 @@
+        fill='#6075AA'
+      />
+      <g clipPath='url(#clip0_63_483)'>
+        <path
+          d='M14.7847 17.2499C16.0273 17.2499 17.0347 16.2426 17.0347 14.9999C17.0347 13.7573 16.0273 12.7499 14.7847 12.7499C13.542 12.7499 12.5347 13.7573 12.5347 14.9999C12.5347 16.2426 13.542 17.2499 14.7847 17.2499Z'
+          stroke='#6075AA'
+          strokeWidth='2'
+          strokeLinecap='round'
+          strokeLinejoin='round'
+        />
+        <path
+          d='M14.7847 26.2499C16.0273 26.2499 17.0347 25.2426 17.0347 23.9999C17.0347 22.7573 16.0273 21.7499 14.7847 21.7499C13.542 21.7499 12.5347 22.7573 12.5347 23.9999C12.5347 25.2426 13.542 26.2499 14.7847 26.2499Z'
+          stroke='#6075AA'
+          strokeWidth='2'
+          strokeLinecap='round'
+          strokeLinejoin='round'
+        />
+        <path
+          d='M14.7847 35.2499C16.0273 35.2499 17.0347 34.2426 17.0347 32.9999C17.0347 31.7573 16.0273 30.7499 14.7847 30.7499C13.542 30.7499 12.5347 31.7573 12.5347 32.9999C12.5347 34.2426 13.542 35.2499 14.7847 35.2499Z'
+          stroke='#6075AA'
+          strokeWidth='2'
+          strokeLinecap='round'
+          strokeLinejoin='round'
+        />
+        <path
+          d='M20.0347 14.9999H35.0347'
+          stroke='#6075AA'
+          strokeWidth='2'
+          strokeLinecap='round'
+          strokeLinejoin='round'
+        />
+        <path
+          d='M20.0347 23.9999H35.0347'
+          stroke='#6075AA'
+          strokeWidth='2'
+          strokeLinecap='round'
+          strokeLinejoin='round'
+        />
+        <path
+          d='M20.0347 32.9999H35.0347'
+          stroke='#6075AA'
+          strokeWidth='2'
+          strokeLinecap='round'
+          strokeLinejoin='round'
+        />
+      </g>
+      <defs>
+        <clipPath id='clip0_63_483'>
+          <rect width='24' height='24' fill='white' transform='translate(11.7847 11.9999)' />
+        </clipPath>
+      </defs>
+    </svg>
+  )
+}
+
+type UseStepIconResult = { [k: string]: string }
+
+export function useStepIcon(currentStepIdx: number, activeStepIdx: number): UseStepIconResult {
+  function getStepState(currentStepIdx: number, activeStepIdx: number): StepState {
+    if (currentStepIdx < activeStepIdx) {
+      return StepState.Visited
+    }
+
+    if (currentStepIdx > activeStepIdx) {
+      return StepState.ToBeVisited
+    }
+
+    return StepState.Current
+  }
```

Comment:
This function allows to know what is the state of the current step while compared to the active step.

Perhaps we could pick better names:
```suggestion
  function getStepState(stepIdx: number, currentStepIdx: number): StepState {
    if (stepIdx < currentStepIdx) {
      return StepState.Visited
    }

    if (stepIdx > currentStepIdx) {
      return StepState.ToBeVisited
    }

    return StepState.Current
  }
```